gitter.py

```diff
@@ -216,6 +224,40 @@ def __unicode__(self):
 
     __str__ = __unicode__
 
+class GitterRoomThread(threading.Thread):
+    def __init__(self, room, backend):
+        super().__init__()
+        self.room = room
+        self.backend = backend
+
+    def run(self):
+        self.room.joined = True
+        log.debug("thread for %s started" % self.room.idd)
+        while True:
+            self.stream()
+
+    def stream(self):
+        r = self.backend.streamAPIRequest('rooms/%s/chatMessages' % self.room.idd)
+        log.debug("connected %s" % self.room.name)
+
+        try:
+            for line in r.iter_lines(chunk_size=1):  # it fails with anything else than 1.
+                if line.strip():
+                    json_message = json.loads(line.decode('utf-8'))
+                    from_user = json_message['fromUser']
+                    log.debug("Raw message from room %s: %s" % (self.room.name, json_message))
+                    if self.room._uri == from_user['url']:
+                        m = Message(json_message['text'])
+                        m.to = self.backend.bot_identifier
+                    else:
+                        m = Message(json_message['text'])
+                        m.to = self.room
+                    m.frm = GitterRoomOccupant.build_from_json(self.room, from_user)
+                    self.backend.callback_message(m)
+                else:
+                    log.debug('Received keep-alive on %s', self.room.name)
+        except:
+            return
```

Comment:
Log the exception here otherwise it might not work and the user will never know why